sdk/common/build.gradle.kts

```diff
@@ -1,9 +1,11 @@
+import io.opentelemetry.gradle.OtelVersionClassPlugin
+
 plugins {
   id("otel.java-conventions")
   id("otel.publish-conventions")
-
   id("otel.animalsniffer-conventions")
 }
+apply<OtelVersionClassPlugin>()
```

Comment:
If we do this for `:sdk:common`, should we do it everywhere?

Reply:
I only wanted to do this in the one place where it's used. If there are other places that ever need to reference the class, then those modules can apply the plugin. I don't think we need it floating around every module. Feels bloaty.

Reply:
And those will still continue work, I didn't _remove_ the properties behavior. Are you suggesting that we should tackle those all in this same PR?

Reply:
> Are you suggesting that we should tackle those all in this same PR?

If we have this pattern, it offers a pretty clean way of accessing the version. Adding a plugin to the gradle build is cleaner than copy / pasting the code to read from the resource. 

Doesn't have to be in this PR, but if we like this pattern I think we should standardize on it.